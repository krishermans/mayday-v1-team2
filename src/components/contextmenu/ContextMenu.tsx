import * as React from "react";
import { Menu, MenuProps, MenuTrigger } from "@fluentui/react-components";
import ContextMenuTrigger from "./ContextMenuTrigger";
import { createContextMenu } from "./ContextMenu.utils";

import type { ContextMenuProps } from "./ContextMenu.types";


const ContextMenu: React.FC<ContextMenuProps> = ({ trigger, menuItems }) => {

    const [open, setOpen] = React.useState(false);
    const onOpenChange: MenuProps["onOpenChange"] = (_, data) => {
        setOpen(data.open);
    };

    return (
        <Menu open={open} onOpenChange={onOpenChange} openOnContext={true}>
            <MenuTrigger disableButtonEnhancement>
                <ContextMenuTrigger>
                    {trigger}
                </ContextMenuTrigger>
            </MenuTrigger>
            {createContextMenu(menuItems)}
        </Menu>
    );

};

export default ContextMenu;