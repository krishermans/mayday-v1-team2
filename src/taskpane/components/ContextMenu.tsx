import * as React from "react";

import {
    Button,
    Menu,
    MenuTrigger,
    MenuList,
    MenuItem,
    MenuPopover,
} from "@fluentui/react-components";

import type {
    MenuProps,
    MenuTriggerChildProps,
} from "@fluentui/react-components";

const CustomMenuTrigger = React.forwardRef<
    HTMLDivElement,
    Partial<MenuTriggerChildProps>
>((props, ref) => {
    return (
        <div {...props} ref={ref}>
            Custom Trigger
        </div>
    );
});

export const CustomTrigger = ({menu1, menu2, menu3}) => {
    const [open, setOpen] = React.useState(false);
    const onOpenChange: MenuProps["onOpenChange"] = (_e, data) => {
        setOpen(data.open);
    };

    return (
        <Menu open={open} onOpenChange={onOpenChange} openOnContext={true}>
            <MenuTrigger disableButtonEnhancement>
                <CustomMenuTrigger />
            </MenuTrigger>

            <MenuPopover>
                <MenuList>
                    <MenuItem>{menu1}</MenuItem>
                    <MenuItem>{menu2}</MenuItem>
                    <MenuItem>{menu3}</MenuItem>
                </MenuList>
            </MenuPopover>
        </Menu>
    );
};



// export const AligningWithIcons = ({title, menu1, menu2, menu3}) => (
//     <Menu hasIcons openOnContext={true}>
//         <MenuTrigger disableButtonEnhancement>
//             <Button>{title}</Button>
//         </MenuTrigger>
//         <MenuPopover>
//             <MenuList>
//                 <MenuItem>{menu1}</MenuItem>
//                 <MenuItem >{menu2}</MenuItem>
//                 <MenuItem>{menu3}</MenuItem>
//             </MenuList>
//         </MenuPopover>
//     </Menu>
// );