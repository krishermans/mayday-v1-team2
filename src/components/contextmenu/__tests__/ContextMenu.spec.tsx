import React from "react";
import { fireEvent, render } from "@testing-library/react";
import ContextMenu from "../ContextMenu";


describe("ContextMenu Test Suite", () => {

    const trigger = <h1 role="trigger">Trigger</h1>;
    const menuItems = [{ handler: jest.fn(), label: "Label 1" }];
    const renderComponent = () => (render(<ContextMenu trigger={trigger} menuItems={menuItems}/>));

    test("Initial render", () => {
        const { baseElement } = renderComponent();

        expect(baseElement).toMatchSnapshot();
    });

    test("Initially open = false", () => {
        const { queryByRole } = renderComponent();

        expect(queryByRole("menu")).not.toBeInTheDocument();
    });

    test("onOpenChange", () => {
        const { getByRole } = renderComponent();

        fireEvent.contextMenu(getByRole("trigger"));

        expect(getByRole("menu")).toBeInTheDocument();
    });

});