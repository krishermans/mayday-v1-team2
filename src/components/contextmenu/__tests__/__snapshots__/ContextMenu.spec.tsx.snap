// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu Test Suite Initial render 1`] = `
<body>
  <div>
    <div
      data-tabster="{"restorer":{"type":1}}"
      id="menur0"
    >
      <h1
        role="trigger"
      >
        Trigger
      </h1>
    </div>
  </div>
</body>
`;
